### PR TITLE
Add attention instead of TDID correlation

### PR DESCRIPTION
**Pull request by mike-shvets**
Core code is in attention.py.
Config is updated with ATTENTION_ENABLED, ATTENTION_COMBINATION_MODE, and ATTENTION_NORMALIZE_MODE.
TDID class is updated to optionally include attention.

### CODE REVIEW

File: model_defs/attention.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+        # B x C x HW
+        y = y.permute(0, 2, 1).contiguous()
+        y = y.view(batch_size, self.inter_channels, *x.size()[2:])
+        W_y = self.W(y)
+        if self.combination_mode == "stack-reduce":
+            z = torch.cat([W_y, x], dim=1)
+            z = self.output_reduce(z)
+        elif self.combination_mode == "attention-only":
+            z = W_y
```
Comment (mike-shvets) [inline model_defs/attention.py:123]: Given that in TDID final version image features do not contribute much, I think it's good to try and train this version with "average" aggregation

File: configs/configAVD2.py [hunk 2]
```diff
@@ -7,32 +7,38 @@ class Config():
...
     TEST_OUTPUT_DIR = os.path.join(DATA_BASE_DIR, 'TestOutputs/')
     TEST_GROUND_TRUTH_BOXES = os.path.join(DATA_BASE_DIR, 'GT/AVD_split2_test.json')
-    VAL_GROUND_TRUTH_BOXES = os.path.join(DATA_BASE_DIR ,'GT/AVD_part3_val.json')
+    VAL_GROUND_TRUTH_BOXES = os.path.join(DATA_BASE_DIR ,'GT/AVD_split2_test.json')
 
+    ATTENTION_ENABLED = True
+    # "sum", or "stack-reduce", or "attention-only"
+    ATTENTION_COMBINATION_MODE = "attention-only"
+    # "average" or "softmax"
+    ATTENTION_NORMALIZE_MODE = "average"
```
Comment (mike-shvets) [inline configs/configAVD2.py:25]: I am adding these three flags. First is optional, as main code uses getattr and defaults to False. The other two are required in case first one is True.

Let's try to train a couple settings on the empty machines. Suggested:
1) sum + softmax
2) sum + average
3) attention-only + average
4) stack-reduce + softmax
5) stack-reduce + average

I don't believe too much in softmax version, upon our discussion
